Client/utils/appAPI.js

```diff
@@ -0,0 +1,273 @@
+import axios from 'axios';
+
+// const AppActions = require('../actions/AppActions');
+import AppActions from '../actions/AppActions';
+
+module.exports = {
+  registerNewUser(user) {
+    console.log(user);
+    axios.post('/users/signup', {
+      email: user.email,
+      password: user.password,
+      userName: user.username,
+      photoURL: user.profilePic,
+      phoneNo: user.phoneNo
+    }).then((response) => {
+      const authuser = {
+        id: response.data.user.uid,
+        email: user.email,
+        profilePic: response.data.user.photoURL,
+        displayName: response.data.user.displayName,
+        phoneNo: response.data.user.phoneNumber,
+        isAuthenticated: true
+      };
+      AppActions.receiveSuccess(response.data.message);
+      AppActions.receiveAuthenticatedUser(authuser);
+      AppActions.receiveErrors('');
+    })
+  .catch((error) => {
+    console.log(error);
+    AppActions.receiveErrors(error.message);
+  });
+  },
+
+  updateUserProfile(user) {
+    // console.log(user);
+    axios.post('/users/updateUserProfile', {
+      userName: user.username,
+      photoURL: user.profilePic,
+      phoneNo: user.phoneNo
+    }).then((response) => {
+      AppActions.receiveSuccess(response.data.message);
+      AppActions.receiveErrors('');
+    })
+  .catch((error) => {
+    AppActions.receiveErrors(error.message);
+  });
+  },
+
+  signinUser(user) {
+   // console.log(user);
+    axios.post('/users/signin', {
+      email: user.email,
+      password: user.password
+    }).then((response) => {
+      // console.log(response);
+      const authuser = {
+        id: response.data.user.uid,
+        email: user.email,
+        profilePic: response.data.user.photoURL,
+        displayName: response.data.user.displayName,
+        phoneNo: response.data.user.phoneNumber,
+        isAuthenticated: true
+      };
+      AppActions.receiveSuccess(response.data.message);
+      AppActions.receiveAuthenticatedUser(authuser);
+      AppActions.receiveErrors('');
+      console.log(authuser);
+    })
+  .catch((error) => {
+    AppActions.receiveErrors(error.message);
+    // console.log(error);
+    // console.log(user);
+  });
+  },
+
+  resetPassword(emailAddress) {
+   // console.log(user);
+    axios.post('/users/passwordReset', {
+      emailAddress
+    }).then((response) => {
+      AppActions.receiveSuccess(response.data.message);
+      AppActions.receiveErrors('');
+     // console.log(authuser);
```

Comment:
You don't want commented out code in production. If it's no longer useful, delete it.